Update Program.cs

dar10 opened the pull request:
The original program.cs returned the following error:
NotFound error occurred: Microsoft.Azure.Cosmos.CosmosException : Response status code does not indicate success: 404 Substatus: 0 Reason: ({
  "Errors": [
    "Resource Not Found"
  ]
}).
StatusCode = NotFound;
SubStatusCode = 0;

## Purpose
<!-- Describe the intention of the changes being proposed. What problem does it solve or functionality does it add? -->
* ...

## Does this introduce a breaking change?
<!-- Mark one with an "x". -->
```
[ ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code
<!-- Add steps to run the tests suite and/or manually test -->
```
```

## What to Check
Verify that the following are valid
* ...

## Other Information
<!-- Add any other helpful information that may be needed here. -->

dar10 committed:
Update Program.cs

The original program.cs returned the following error:
NotFound error occurred: Microsoft.Azure.Cosmos.CosmosException : Response status code does not indicate success: 404 Substatus: 0 Reason: ({
  "Errors": [
    "Resource Not Found"
  ]
}).
StatusCode = NotFound;
SubStatusCode = 0;

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/signed)](https://cla.opensource.microsoft.com/Azure-Samples/cosmos-dotnet-core-getting-started?pullRequest=2) <br/>All CLA requirements met.

ealsur commented on CosmosGettingStartedDotnetCoreTutorial/Program.cs:
```diff
@@ -118,21 +118,8 @@ private async Task AddItemsToContainerAsync()
-                Console.WriteLine("Item in database with id: {0} already exists\n", andersenFamilyResponse.Resource.Id);
-            }
+            // Upsert the item, i.e. if it exists it gets updated, otherwise it gets created
+            ItemResponse<Family> andersenFamilyResponse = await this.container.UpsertItemAsync<Family>(andersenFamily, new PartitionKey(andersenFamily.LastName));
```
I believe this beats the purpose of the original code, while I agree that Upsert solves the exception, the code was showing how to handle a scenario where an item might Not exist.

Possible try/catch CosmosException when StatusCode == NotFound

dar10 replied:
@ealsur Thanks for the feedback. I added try catch semantics to the code. Keep in mind that the method Container.ReadItemAsync<T> returns an exception when the code runs. Therefore, since the Cosmos Database and its container are empty the catch block will always execute and the items will get created in each catch clause.

ealsur reviewed (changes requested):
Any code fix should not break the intent of the original code, rather fix it by adding the explicit catch

dar10 committed:
Update Program.cs

Since the Method Container.ReadItemAsync<T> will throw an exception "Resource Not Found (404)" I added try-catch semantics to catch it and act accordingly.
